### PR TITLE
feat: support write method

### PR DESCRIPTION
**Pull request by hiiiik**

### CODE REVIEW

File: include/opengemini/impl/enc/LineProtocolEncoder.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+{
+    AppendPoint(point);
+    return os_.str();
+}
+
+OPENGEMINI_INLINE_SPECIFIER
+std::string LineProtocolEncoder::Encode(const std::vector<Point>& points)
+{
+    if (points.empty()) {
+        throw Exception(errc::LogicErrors::InvalidArgument,
```
Comment (shoothzj) [inline include/opengemini/impl/enc/LineProtocolEncoder.cpp:56]: This should be sync with other languages

File: include/opengemini/impl/enc/LineProtocolEncoder.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            else if constexpr (std::is_same_v<T, int64_t>) {
+                Append(innerValue);
+                Append(ELEMENT_INT);
+            }
+            else if constexpr (std::is_same_v<T, bool>) {
+                Append(innerValue ? ELEMENT_TRUE : ELEMENT_FALSE);
+            }
+            else if constexpr (std::is_same_v<T, double>) {
+                Append(innerValue);
+            }
```
Comment (shoothzj) [inline include/opengemini/impl/enc/LineProtocolEncoder.cpp:170]: shoud we add a default error? like

```c++
else {
    static_assert(always_false<T>::value, "Unhandled type in std::variant.");
}
```

File: include/opengemini/impl/enc/LineProtocolEncoder.cpp (hunk 1 above)
Comment (hiiiik) [reply include/opengemini/impl/enc/LineProtocolEncoder.cpp]: Now we allow user to pass an empty vector of points that the library will neither throw an exception nor invoke any request to server for it.

File: include/opengemini/impl/enc/LineProtocolEncoder.cpp (hunk 2 above)
Comment (hiiiik) [reply include/opengemini/impl/enc/LineProtocolEncoder.cpp:161]: Good advice, I will add a runtime checking in the else branch.

**Review by shoothzj — APPROVED** [review]

**Pull request merged**